src/nl/rubensten/texifyidea/inspections/bibtex/BibtexDuplicateBibliographyInspection.kt

```diff
@@ -31,20 +31,33 @@ open class BibtexDuplicateBibliographyInspection : TexifyInspectionBase() {
     override fun inspectFile(file: PsiFile, manager: InspectionManager, isOntheFly: Boolean): MutableList<ProblemDescriptor> {
         val descriptors = descriptorList()
 
+        val groupedIncludes = mutableMapOf<String, MutableList<LatexCommands>>()
+
         LatexIncludesIndex.getItemsInFileSet(file).asSequence()
                 .filter { it.name == "\\bibliography" || it.name == "\\addbibresource" }
-                .groupBy { it.requiredParameters.getOrNull(0) }
-                .filter { it.key != null && it.value.size > 1 }
-                .flatMap { it.value }
                 .forEach {
-                    descriptors.add(manager.createProblemDescriptor(
-                            it,
-                            TextRange(0, it.requiredParameter(0)?.length!!).shiftRight(it.commandToken.textLength + 1),
-                            "Bibliography file is included multiple times",
-                            ProblemHighlightType.GENERIC_ERROR,
-                            isOntheFly,
-                            RemoveOtherCommandsFix(it.requiredParameter(0)!!)
-                    ))
+                    for (fileName in it.includedFileNames() ?: emptyList()) {
+                        groupedIncludes.getOrPut(fileName) { mutableListOf() }.add(it)
+                    }
+                }
+
+        groupedIncludes.asSequence()
+                .filter { it.value.size > 1 }
+                .forEach {
+                    val fileName = it.key
+                    val commands = it.value.distinct()
+                    for (command in commands) {
+                        if (command.containingFile != file) continue
+
+                        descriptors.add(manager.createProblemDescriptor(
+                                command,
+                                TextRange(0, command.requiredParameter(0)?.length!!).shiftRight(command.commandToken.textLength + 1),
```

Comment:
Can this go wrong when `command` has no required parameters?

Reply:
Will never happen, because `command` must have a required parameter since `command` is grouped by the contents of the required parameter in the first place